mongofiles/options.go

```diff
@@ -65,16 +64,9 @@ func ParseOptions(rawArgs []string) (Options, error) {
 	opts.WriteConcern = wc
 
 	// set ReadPreference
-	if inputOpts.ReadPreference != "" {
-		opts.ReadPreference, err = db.ParseReadPreference(inputOpts.ReadPreference)
-		if err != nil {
-			return Options{}, fmt.Errorf("error parsing --readPreference: %v", err)
-		}
-	} else {
-		opts.ReadPreference, err = readpref.New(readpref.NearestMode)
```

Comment:
I think the nearest read preference was being used to mock a direct connection. Afaik, the default read preference should be primary. The logic for creating a direct connection is handled in the new common code.